util/Pending.h

```diff
@@ -135,7 +135,10 @@ namespace Pending {
         bool success = true;
         auto result = parser(arg1, path, success);
         auto promise = std::promise<decltype(parser(arg1, path, std::declval<bool&>()))>();
-        promise.set_value(std::move(result));
+        if (success)
+            promise.set_value(std::move(result));
+        else
+            promise.set_exception(std::make_exception_ptr(std::runtime_error(path.string() + " failed to parse, see logs for more info")));
```

Comment:
Maybe don't put english text into this and rather return an exception containing just the failed filename. Will make show in the UI with a localize message easier.